src/components/LoginContent.js

```diff
@@ -0,0 +1,225 @@
+        this.onRegisterLoginLinkClick = this.onRegisterLoginLinkClick.bind(this);
+        this.triggerPrimaryPasswordVisilibity = this.triggerPrimaryPasswordVisilibity.bind(this);
+        this.triggerConfirmationPasswordVisilibity = this.triggerConfirmationPasswordVisilibity.bind(this);
+        this.onForgotPasswordClick = this.onForgotPasswordClick.bind(this);
+        this.comparePasswords = this.comparePasswords.bind(this);
+    }
+
+    payload = {
+        fullName: this.fullName,
+        email: this.email,
+        password: this.password,
+    } 
+
+    onRegisterLoginLinkClick() {
+      // In order to animate the next state change
+      LayoutAnimation.configureNext(LayoutAnimation.Presets.linear);
+      this.setState({ isInRegister : !this.state.isInRegister });
+    }
+
+    onForgotPasswordClick() {
+      this.props.navigation.navigate('ForgotPasswordScreen');
+    }
+
+    handleActionButton() {
+        if (this.state.isInRegister) {
+            // Handle register functions
+            // JSON data to send when registering
+            let newUserCredentials = {
+                "name": this.payload.fullName,
+                "email": this.payload.email,
+                "password": this.payload.password
+            }
+            
+            axios
+                .post('http://34.125.37.12:8080/api/register', newUserCredentials)
+                .then(
+                    response => {
+                        console.log(response.data);
+                        this.props.navigation.navigate('RegisterScreen', {
+                            userPersonalEmail: this.payload.email,
+                          });
+                        // TODO: Save jwt in AsyncStorage
+                    }
+                )
+                .catch(
+                    error => {
+                        console.log(error)
+                    }
+                )
+                
+        } else {
+            // Handle login functions
+            // JSON data to send when logging in
+            let credentials = {
+                "email": this.payload.email,
+                "password": this.payload.password
+            }
+
+            axios
+                .post('http://34.125.37.12:8080/api/login', credentials)
+                .then(
+                    response => {
+                        console.log(response.data)
+                        this.setState({ isLoggedIn: response.data.token ? true : false }, () => {
+                            this.props.handleLogin(this.state.isLoggedIn)
+                        });
+                        // TODO: Save jwt in AsyncStorage
+                    }
+                )
+                .catch(
+                    error => {
+                        console.log(error)
+                    }
+                )
+                
+        }
+    }
+
+    triggerPrimaryPasswordVisilibity() {
+        this.setState({ hidePrimaryPassword : !this.state.hidePrimaryPassword });
```

Comment:
This is seen in a lot of places. According to the [docs](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous), if the state depends on the previous state, you must use the arrow function.
```this.setState(oldState => ({ hidePrimaryPassword: !oldState. hidePrimaryPassword }));```